### ISSUE TITLE
idea: command line switch for headless update

### ISSUE THREAD

**Issue by illwieckz** [Unvanquished/updater#14]
For server owners, would be cool to be able to do:

./updater --no-gui --install-path=/some/where

I don't know yet if `--no-gui` would just check for update and download things or must it run the game at the end too. In the later case, perhaps add a `--no-run` option, to let server owners having one install for multiple servers being sure they keep full control of what they do.

The `--install-path` would be optional. Once the config file is written, the install path would be reused like in gui mode. Wa can also add a `--no-conf` switch to not write the config file, it would be useful for benchmarking software, allowing them to install unvanquished for benchmarking purpose without having to mess with user data path (Let's make phoronix benchmarking us again).

_Edit: the `--install-path` would be optionally usable by game managers like [lutris](http://lutris.net/) or things like that, so it would be cool to have it doing things on gui mode too._

**Comment by DolceTriade** [comment]
I feel like this is overkill for such a task. We have ./download_pk3_torrent.sh and such that accomplish the same thing. If we want to do headless, we should ship a static aria2c binary with a batch/shell script instead.

**Comment by illwieckz** [comment]
Static aria2c binary is not a bad idea but it's yet another project to maintain, debug, etc.

**Comment by illwieckz** [comment]
For testing and debugging purpose I would like to have those options even in graphical mode: `--no-config-write`, `--auto-download`, `--start-after-download` (see #48), `--download-only`, that's while a `--no-gui` option or a stripped-down build, console only variant may be a good idea.

**Comment by DolceTriade** [comment]
We also probably don't want to require QT for such a build, hence the bash script

**Comment by ghost** [comment]
I would not use a qt application on my server to update when it's simple enough to do it manually.

**Comment by necessarily-equal** [comment]
I don't think this use case for the updater is worthy of our time — when we add another server, we want to to be unique, and that means it must have its own VM files, and then it basically needs a custom install anyway.

**Comment by ghost** [comment]
Well, there's a subtle nuance to put to @necessarily-equal though: we prefer servers to have unique settings and/or content *for their geographic area*.
So far, (almost) all servers only notably host SK5 bots which is bot good and bad. but, that's a different topic I believe. Maybe a tool to generate config with presets could be nice though, since unv is fairly poorly documented. I still think, documenting and have complete config example file would be a better spend of time.
